### ISSUE TITLE
Fix pos-tip setup in `master`

### ISSUE THREAD

**Issue by lunaryorn** [syl20bnr/spacemacs#4181]
Please backport the flycheck-pos-tip setup from `develop` to a stable release. 

When flycheck-pos-tip changed to a minor mode a month ago, `develop` was updated, but the change was apparently not back-ported to `master`.  Consequently all stable Spacemacs releases have an incomplete setup for flycheck-pos-tip which leads to [some feature missing](https://github.com/flycheck/flycheck-pos-tip/issues/12) and confuses users.

**Comment by syl20bnr** [comment]
Thank you for the report :+1: 
I'll do a release this evening.

**Comment by StreakyCobra** [comment]
@syl20bnr Just to be sure, hotfix release? Or do I have to stress releasing the bepo layer update :-p

**Comment by syl20bnr** [comment]
Done :-)
@StreakyCobra Bepo layer does not exist in 0.104.7. If everything goes well we should have 0.105 shipped for Christmas.

**Comment by haroldcarr** [comment]
I just updated to 0.104.7.  When I do (use-package flycheck) I get:

    Debugger entered--Lisp error: (void-function flycheck-pos-tip-mode)
      (flycheck-pos-tip-mode)
      (lambda nil (flycheck-pos-tip-mode))()
      eval-after-load-helper("/Users/carr/.emacs.d/elpa/flycheck-20150921.947/flycheck.elc")
      run-hook-with-args(eval-after-load-helper "/Users/carr/.emacs.d/elpa/flycheck-20150921.947/flycheck.elc")
      do-after-load-evaluation("/Users/carr/.emacs.d/elpa/flycheck-20150921.947/flycheck.elc")
      require(flycheck nil noerror)
      (not (require (quote flycheck) nil (quote noerror)))

**Comment by lunaryorn** [comment]
@haroldcarr Please update your packages, and consider updating frequently.  The package is almost three months old, which is a rather long time in Emacs land.

**Comment by haroldcarr** [comment]
I guess I was under the mistaken assumption that spacemacs was curating the package versions.

When you say "update your packages" - do I just do an update all or what specifically?

**Comment by lunaryorn** [comment]
Indeed, the infrastructure that we (as a community) currently have does not allow for this.  We lack the tools to manage package versions as strictly as in other environment such as Ruby or Haskell.

I'm sorry for having been unclear in my comment; you may update `flycheck-pos-tip` specifically, but I would strongly recommend you to just use the "Update" button in the Spacemacs buffer (the one that appears when you start Spacemacs).  That is safer, and—given the age of `flycheck-pos-tip`—I think you would generally profit from a general update of all packages.

**Comment by haroldcarr** [comment]
Now I see.  I thought when I did an upgrade (say from 0.104.6 to 0.104.7 like I just did) that it updated package versions too.   Now I understand that the Update button (which I suppose it the same a list-packages, mark upgradable packages, execute actions) is to keep packages up-to-date and "upgrade" is to upgrade to the latest spacemacs infrastructure.

Question: would you recommend to Update (i.e., packages) only when one upgrades spacemacs or even in between upgrades?

ps: loving spacemacs (and I'm a long-time emacs user)

**Comment by lunaryorn** [comment]
I do not know what Spacemacs recommends.  For my part, I follow the `develop` branch of Spacemacs and update packages very frequently.

**Comment by StreakyCobra** [comment]
@haroldcarr It may be a good idea to update packages at the same time as updating Spacemacs, because there could have been some changes in Spacemacs that support only the new versions of some packages. It's not always the case nor mandatory though, especially for minor release.

Updates usually bring bugfixes and/or new features, but sadly they can also bring new bugs. So:

- If you want a **strongly stable** text editor, use `master` and only update packages when something is broken, and on new major Spacemacs releases. 

  *Drawback: the jump can be quite big when an update will be required.*

- If you want a **stable** text editor, use `master` and update packages from time to time (also on new Spacemacs releases), when you know you will have time to test Spacemacs to detect some eventual new bugs (allowing you to rollback the packages updates if needed).

  *Probably the safest way.*

- If you want a **fancy** text editor with some stability, use `develop` and update Spacemacs branch and packages only when you know you will have time to test Spacemacs to detect some eventual new bugs (allowing you to rollback the packages updates and to checkout the previously working `develop` commit).

  *For people wanting a fancy editor, or requiring features from `develop`.*

- If you want **the fanciest** text editor, use `develop` and update Spacemacs branch and packages every day.

  *Useful for contributors to catch new bugs soon enough :-) and for people who like to live dangerously.*

**Comment by syl20bnr** [comment]
@StreakyCobra thanks for the use cases list!

@haroldcarr
TL;DR Update Spacemacs only when you want to update your packages, recommended workflow:

Spacemacs has only an update/rollback mechanism for now, if you have a working Spacemacs then you can consider updating your packages often and rollback as need to go back to a working state. But we are still lacking an important mechanism which to be able to bundle a set of stable packages for a given version of Spacemacs so currently a user installing Spacemacs _must_ retrieve the last version of all packages. This is a very uncomfortable situation and we try to mitigate this by release minor version to remain compatible with the last versions of packages. It means for you that as long as you don't update your packages you don't have to update Spacemacs. So if I have a recommendation it might be a surprising one since it would be to not update Spacemacs if you don't want to update your packages. Here is a diagram for the worst cases:
```
update packages --- works ---> update spacemacs ----- does not work ----> revert spacemacs version
      |____does not work ----> update spacemacs ----- does not work ----> rollback packages --^
```
